tutor/hooks/consts.py

```diff
@@ -171,21 +187,27 @@ def your_filter(items):
     #:      rendered at runtime with the user configuration. Thus, the image tag could
     #:      be ``"{{ DOCKER_REGISTRY }}/myimage:{{ TUTOR_VERSION }}"``.
     #:    - ``args`` is a list of arguments that will be passed to ``docker build ...``.
-    #: :parameter dict config: user configuration.
-    IMAGES_BUILD = filters.get("images:build")
+    #: :parameter Config config: user configuration.
+    IMAGES_BUILD: "Filter[t.List[t.Tuple[str, t.Tuple[str, ...], str, t.Tuple[str, ...]]], [Config]]" = filters.get(
+        "images:build"
+    )
 
     #: List of images to be pulled when we run ``tutor images pull ...``.
     #:
     #: :parameter list[tuple[str, str]] tasks: list of ``(name, tag)`` tuples.
     #:
     #:    - ``name`` is the name of the image, as in ``tutor images pull myimage``.
     #:    - ``tag`` is the Docker tag that will be applied to the image. (see :py:data:`IMAGES_BUILD`).
-    #: :parameter dict config: user configuration.
-    IMAGES_PULL = filters.get("images:pull")
+    #: :parameter Config config: user configuration.
+    IMAGES_PULL: "Filter[t.List[t.Tuple[str, str]], [Config]]" = filters.get(
+        "images:pull"
+    )
 
     #: List of images to be pushed when we run ``tutor images push ...``.
     #: Parameters are the same as for :py:data:`IMAGES_PULL`.
-    IMAGES_PUSH = filters.get("images:push")
+    IMAGES_PUSH: "Filter[t.List[t.Tuple[str, str]], [Config]]" = filters.get(
+        "images:push"
+    )
```

Comment:
I do not know why I didn't bother adding type definitions below this line. I guess I must have been tired. I'll add the definitions now.